src/Microsoft.AspNetCore.SignalR.Client.Core/InvocationRequest.cs

```diff
@@ -78,26 +77,21 @@ public Streaming(CancellationToken cancellationToken, Type resultType, string in
 
             public ReadableChannel<object> Result => _channel.In;
 
-            public override void Complete(HubMessage message)
+            public override void Complete(CompletionMessage completionMessage)
             {
-                Debug.Assert(message != null, "message is null");
-
-                if (!(message is StreamCompletionMessage streamCompletionMessage))
+                Logger.InvocationCompleted(InvocationId);
+                if (completionMessage.Result != null)
                 {
-                    Logger.ReceivedUnexpectedMessageTypeForStreamCompletion(InvocationId, message.GetType().Name);
-                    // This is not 100% accurate but it is the only case that can be encountered today when running end-to-end
-                    // and this is the most useful message to show to the user.
-                    Fail(new InvalidOperationException($"Streaming hub methods must be invoked with the '{nameof(HubConnection)}.{nameof(HubConnection.StreamAsync)}' method."));
-                    return;
+                    Logger.ReceivedUnexpectedComplete(InvocationId);
+                    _channel.Out.TryComplete(new InvalidOperationException("Server provided a result in a completion response to a streamed invocation."));
```

Comment:
I had to read this twice :)

Reply:
Yeah - it's pretty dense but I just took what we already [had](https://github.com/aspnet/SignalR/commit/8d1d6d03005ea55c646850832de2e0eda53c6ade#diff-d05d3cff4a15053230f359d7e95afe3eL86) before